ee/codegen/src/generators/nodes/bases/base.ts

```diff
@@ -529,6 +530,17 @@ export abstract class BaseNode<
     }
   }
 
+  protected getNodeTrigger(): AstNode | undefined {
+    if (this.nodeData.trigger) {
```

Comment:
to follow up with my comment above, we likely need a comparison here with the default trigger